EpsilonLights/Src/Lights.c

```diff
@@ -20,15 +20,44 @@ void updateLightsTask(void const* arg)
     {
         osDelayUntil(&prevWakeTime, LIGHTS_UPDATE_FREQ);
         headlightsOff = (lightsInputs >> HOFF_INPUT_INDEX) & 1;
-        headlightsLow = (lightsInputs >> HLOW_INPUT_INDEX) & 1;
-        headlightsHigh = (lightsInputs >> HHIGH_INPUT_INDEX) & 1;
+        headlightsLow = (lightsInputs >> HLOW_INPUT_INDEX) & 1; //0b0000x0010
+        headlightsHigh = (lightsInputs >> HHIGH_INPUT_INDEX) & 1; //0b0000x0100
         rightSignal = (lightsInputs >> RSIGNAL_INPUT_INDEX) & 1;
         leftSignal = (lightsInputs >> LSIGNAL_INPUT_INDEX) & 1;
         hazards = (lightsInputs >> HAZARDS_INPUT_INDEX) & 1;
 
         /* UPDATE HEADLIGHTS */
+        if (headlightsLow)
+        {
+            HAL_GPIO_WritePin(HLOW_GPIO_Port, HLOW_Pin, LIGHT_ON);
+        }
+        else
+        {
+            HAL_GPIO_WritePin(HLOW_GPIO_Port, HLOW_Pin, LIGHT_OFF);
+        }
+
+        if (headlightsHigh)
+        {
+            HAL_GPIO_WritePin(HHIGH_GPIO_Port, HHIGH_Pin, LIGHT_ON);
+        }
+        else
+        {
+            HAL_GPIO_WritePin(HHIGH_GPIO_Port, HHIGH_Pin, LIGHT_OFF);
+        }
+
+        if ((headlightsOff) && (headlightsLow || headlightsHigh))
```

Comment:
Imo I think it would make more sense to do 
````
if(headlightsOff)
{
 Turn off lights
}
else if (headlightsHigh && headlightsLow)
{
 Turn off headlights
}
else()
{
 Turn on/off headlights as needed
}
````

That way we don't write the headlights first before checking if headlights are off.